### ISSUE TITLE
Enabling feature "aft_feature_cloudtrail_data_events" causes a failure in step function "aft-account-provisioning-framework".

### ISSUE THREAD

**Issue by brhankel** [aws-ia/terraform-aws-control_tower_account_factory#154]
**Terraform Version & Prov:**

**AFT Version:**
1.3.6

**Terraform Version & Provider Versions**
Please provide the outputs of `terraform version` and `terraform providers` from within your AFT environment

`terraform version`
```
Terraform v1.1.7
```

`terraform providers`
```
+ provider registry.terraform.io/hashicorp/archive v2.2.0
+ provider registry.terraform.io/hashicorp/aws v3.74.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/time v0.7.2
```

**Bug Description**
Enabling feature `aft_feature_cloudtrail_data_events` causes a failure in step function `aft-account-provisioning-framework`.

**To Reproduce**
1. Starting with a fresh Master Account(root) deploy AFT with no features enabled
2. Set `aft_feature_cloudtrail_data_events = true`
3. Create and submit a new Account Request
4. See error on Step Function aft-account-provisioning-framework

**Expected behavior**
An organization CloudTrail for data events should be created in the Master Account(root)

**Related Logs**
Two failures are actually observed both of which are in underlying State Machine `aft-feature-options` 
First one is related to SNS permissions:
```
{
  "Error": "AuthorizationErrorException",
  "Cause": "{\"errorMessage\": \"An error occurred (AuthorizationError) when calling the Publish operation: User: arn:aws:sts::761884181760:assumed-role/aft-enable-cloudtrail-execution-role/aft-enable-cloudtrail is not authorized to perform: SNS:Publish on resource: arn:aws:sns:us-east-1:761884181760:aft-failure-notifications because no identity-based policy allows the SNS:Publish action\", \"errorType\": \"AuthorizationErrorException\", \"stackTrace\": [\"  File \\\"/var/task/aft_enable_cloudtrail.py\\\", line 58, in lambda_handler\\n    notifications.send_lambda_failure_sns_message(\\n\", \"  File \\\"/opt/python/lib/python3.8/site-packages/aft_common/notifications.py\\\", line 48, in send_lambda_failure_sns_message\\n    send_sns_message(\\n\", \"  File \\\"/opt/python/lib/python3.8/site-packages/aft_common/notifications.py\\\", line 31, in send_sns_message\\n    response = client.publish(TopicArn=topic, Message=sns_message, Subject=subject)\\n\", \"  File \\\"/opt/python/lib/python3.8/site-packages/botocore/client.py\\\", line 415, in _api_call\\n    return self._make_api_call(operation_name, kwargs)\\n\", \"  File \\\"/opt/python/lib/python3.8/site-packages/botocore/client.py\\\", line 745, in _make_api_call\\n    raise error_class(parsed_response, operation_name)\\n\"]}"
}
```

If the above is resolved and the Step Function is executed again the following error is then observed where CloudTrail is not enabled at the org level.
```
{
  "Error": "CloudTrailAccessNotEnabledException",
  "Cause": "{\"errorMessage\": \"An error occurred (CloudTrailAccessNotEnabledException) when calling the CreateTrail operation: The request could not be processed because your organization hasn\\u2019t enabled CloudTrail service access. Enable service access for CloudTrail, and then try again. You are signed in with the following AWS account ID: 571511383529\", \"errorType\": \"CloudTrailAccessNotEnabledException\", \"stackTrace\": [\"  File \\\"/var/task/aft_enable_cloudtrail.py\\\", line 51, in lambda_handler\\n    create_trail(ct_session, s3_bucket_name, kms_key_arn)\\n\", \"  File \\\"/opt/python/lib/python3.8/site-packages/aft_common/feature_options.py\\\", line 243, in create_trail\\n    client.create_trail(\\n\", \"  File \\\"/opt/python/lib/python3.8/site-packages/botocore/client.py\\\", line 415, in _api_call\\n    return self._make_api_call(operation_name, kwargs)\\n\", \"  File \\\"/opt/python/lib/python3.8/site-packages/botocore/client.py\\\", line 745, in _make_api_call\\n    raise error_class(parsed_response, operation_name)\\n\"]}"
}
```

**Additional context**
None

**Comment by snebhu3** [comment]
@brhankel thank you for reporting the AuthorizationError for SNS:Publish. We are working on a fix for it with our next release. 

However, for the second error (CloudTrailAccessNotEnabledException), this is expected when trusted access has not been enabled between CloudTrail and Organizations. You could enable service access for CloudTrail and retry.

**Comment by balltrev** [comment]
Just released 1.3.7, that should address the permissions issues introduced by 1.3.6, can you deploy the update and see if that resolves your issue?

**Comment by brhankel** [comment]
Updated to 1.3.7 and confirmed the role has permissions to SNS now. Re-executed the step function with no issues so resolved. Thanks.
